library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/generator/GafferPopElementGeneratorTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.tinkerpop.generator;
+
+import org.junit.jupiter.api.Test;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.mock;
+
+import uk.gov.gchq.gaffer.commonutil.TestGroups;
+import uk.gov.gchq.gaffer.data.element.Edge;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.data.element.Entity;
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopEdge;
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopElement;
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopGraph;
+import uk.gov.gchq.gaffer.tinkerpop.GafferPopVertex;
+
+public class GafferPopElementGeneratorTest {
+
+    @Test
+    public void shouldReturnAGafferPopVertex() {
+        // Given
+        final GafferPopGraph graph = mock(GafferPopGraph.class);
+        final Element element = new Entity.Builder().group(TestGroups.ENTITY).build();
+        final GafferPopElementGenerator generator = new GafferPopElementGenerator(graph, true);
```

Comment:
What is graph used for? what values is the mocked graph meant to return?

Reply:
It is needed to apply the `generator`, see other tests.